scripts/agent_step_zkai_stwo_composition.py

```
@@ -0,0 +1,594 @@
+
+def _checked_stwo_payload(stwo_evidence: dict[str, Any]) -> dict[str, Any]:
+    checked_stwo_evidence_path = stwo_evidence.get("_source_path")
+    if checked_stwo_evidence_path is not None:
+        return checked_stwo_evidence(pathlib.Path(checked_stwo_evidence_path))
+    return _validate_stwo_evidence_payload(stwo_evidence)
+
+
+def _assert_evidence_matches_envelope(stwo_evidence: dict[str, Any], envelope: dict[str, Any]) -> None:
+    cases = stwo_evidence.get("cases")
+    if not isinstance(cases, list) or not cases:
+        raise CompositionError("Stwo benchmark evidence has no cases")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)